core/tests/puppeteer-acceptance-tests/puppeteer-testing-utilities/puppeteer-utils.js

```diff
@@ -80,6 +74,9 @@ module.exports = class puppeteerUtilities {
    */
   async signUpNewUser(userName, signInEmail) {
     await this.signInWithEmail(signInEmail);
+    if (this.page.url() === testConstants.URLs.LearnerDashboard) {
```

Comment:
Yes @seanlip I totally agree ! This is just a patch i did so that the tests do not throw err. Yes, the structure here seems not fitting ! Because essentially we will maintain a single dev server without restarting it for each tests. It will run for the entireity of all tests. The lines which i added above is to address the issue of page redirections. When our user is already signed in, we will be straightaway redirected to the learner dashboard. In case we aren't registered with our email we will be asked to input a username and then proceed to the learner dashboard. I think we need to restructure this a bit and the naming for this function needs some tweaking too. cc @Shivkant-Chauhan

Reply:
Yes @seanlip I totally agree ! This is just a patch i did so that the tests do not throw err. Yes, the structure here seems not fitting ! Because essentially we will maintain a single dev server without restarting it for each tests. It will run for the entireity of all tests. The lines which i added above is to address the issue of page redirections. When our user is already signed in, we will be straightaway redirected to the learner dashboard. In case we aren't registered with our email we will be asked to input a username and then proceed to the learner dashboard. I think we need to restructure this a bit and the naming for this function needs some tweaking too. cc @Shivkant-Chauhan

Reply:
@vojtechjelinek will tell you more about this.. we have discussed it earlier that for each test we will restart the google cloud server and firebase emulator server (instead of restarting the whole dev server), and this would be efficient as it would take upto 30-40 seconds

Reply:
Note that there should be separate methods for signing up as a new user and logging in as an existing user, and the right one should be called at the right time.

As Shivkant says, it is also fine to clear and restart the server after each test. (Please also confirm that the server data still gets cleared if a test fails midway.)